flink-runtime/src/test/java/org/apache/flink/runtime/concurrent/FutureUtilsTest.java

```diff
@@ -235,6 +237,57 @@ public void testRetryWithDelayCancellation() {
 		assertTrue(scheduledFuture.isCancelled());
 	}
 
+	/**
+	 * Tests that the operation could be scheduled with expected delay.
+	 */
+	@Test
+	public void testScheduleAsyncDelay() throws Exception  {
+		final Time delay = Time.milliseconds(500L);
+		final long start = System.currentTimeMillis();
+		final AtomicInteger updateValue = new AtomicInteger(0);
+
+		CompletableFuture<Integer> scheduleAsyncFuture = FutureUtils.scheduleAsync(
+			() -> updateValue.addAndGet(ThreadLocalRandom.current().nextInt(100)),
+			delay,
+			TestingUtils.defaultScheduledExecutor());
+
+		int result = scheduleAsyncFuture.get();
+		long completionTime = System.currentTimeMillis() - start;
+
+		assertEquals(updateValue.get(), result);
+		assertTrue("The completion time should larger than scheduled delay time.", completionTime >= delay.toMilliseconds());
+	}
+
+	/**
+	 * Tests that all scheduled tasks are canceled if the scheduled future is being cancelled.
+	 */
+	@Test
+	public void testScheduleAsyncDelayCancellation() {
+		final ManuallyTriggeredScheduledExecutor scheduledExecutor = new ManuallyTriggeredScheduledExecutor();
+
+		CompletableFuture<?> scheduleAsyncFuture = FutureUtils.scheduleAsync(
+			() -> {
+				throw new FlinkRuntimeException("Test Exception");
+			},
+			TestingUtils.infiniteTime(),
+			scheduledExecutor);
+
+		assertFalse(scheduleAsyncFuture.isDone());
+
+		final Collection<ScheduledFuture<?>> scheduledTasks = scheduledExecutor.getScheduledTasks();
+
+		assertFalse(scheduledTasks.isEmpty());
+
+		final ScheduledFuture<?> scheduledFuture = scheduledTasks.iterator().next();
+
+		assertFalse(scheduledFuture.isDone());
```

Comment:
I don't think this assertion is needed because you already test if the future can be cancelled:

```
assertTrue(scheduledFuture.isCancelled());
```

This assertion will fail if the future is finished normally.